connectivity/FEATURE_BLE/include/ble/Gap.h

```diff
@@ -1633,7 +1450,16 @@ class Gap : public StaticInterface<Impl, Gap> {
 
 #if !defined(DOXYGEN_ONLY)
 } // namespace interface
-#endif
+#endif // !defined(DOXYGEN_ONLY)
 } // namespace ble
 
-#endif //BLE_GAP_GAP_H
+#if (BLE_API_IMPLEMENTATION == 1)
+#include "ble/internal/GapImpl.h"
+#else
+#error "please provide alternate BLE implementation"
+#endif
```

Comment:
```suggestion
#include "ble/internal/GapImpl.h"
```
As discussed offline, this include is okay. I guess the standardised header name can be considered part of the "porting interface", and I've seen a few other projects doing this too.

Actually the `BLE_API_IMPLEMENTATION` macro may not be needed. If a user doesn't want Cordio, they can simply undeclare Cordio from the Mbed config so `TARGET_CORDIO` isn't built, and provide their implementation with the same `"ble/internal/GapImpl.h"` file path.

Reply:
yes, totally agreed, this was only to show the porting point but this can be made better in a readme somewhere